pkg/controller/virtualnode.go

```diff
@@ -314,6 +314,9 @@ func vnodeLoggingNeedsUpdate(desired *appmeshv1beta1.VirtualNode, target *aws.Vi
 
 func (c *Controller) handleVNodeDelete(ctx context.Context, vnode *appmeshv1beta1.VirtualNode, copy *appmeshv1beta1.VirtualNode) error {
 	if yes, _ := containsFinalizer(vnode, virtualNodeDeletionFinalizerName); yes {
+		if err := c.deregisterInstancesForVirtualNode(ctx, vnode); err != nil {
+			return err
```

Comment:
Should we still try to delete the virtual node even if instance deregistration fails?  I.e. if cloudmap is down, is it safer to fail and retry on deregistering instances, and leave the virtualnode around until we succeed, or delete the virtualnode immediately without worrying about cleaning up cloudmap?  I'm thinking the former is correct, which is the logic you have, otherwise we potentially leave cloudmap instances around after the virtualnode is gone.

Reply:
yeah former is safer and predictable.